rl/function_approx.py

```diff
@@ -28,6 +31,14 @@ def update(
     ) -> FunctionApprox[X]:
         pass
 
+    @abstractmethod
+    def solve(
```

Comment:
Would be great to add a docstring to this method, especially since it's abstract.

Reply:
Added docstring, as you suggested